File: src/content/blog.ts
import { z, defineCollection, reference } from "astro:content";

const blogCollection = defineCollection({
  type: "content",
  schema: z.object({
    title: z.string(),
    date: z.date().default(() => new Date()),
    slug: z.string(),
    image: z.string().optional(),
    excerpt: z.string().optional(),  // A short summary or teaser for the post.
    content: z.string(),            // The main content of the blog post.
    tags: z.array(z.string()).optional(),
    author: z.string().optional(),
    authorEmail: z.string().optional(),
    relatedProducts: z.array(reference("products")).optional(),
    relatedBundles: z.array(reference("bundles")).optional(),
    isVisible: z.boolean().default(() => true),   // To control the visibility of the post.
    featureOnHomepage: z.boolean().optional(),    // If the post should be featured on the homepage.

    SEOTitle: z.string().optional(),
  SEODescription: z.string().optional(),
  SEOKeywords: z.array(z.string()).optional(),
  }),
});

export default blogCollection;

File: cli/models/blog/blog.go
package blog

import (
	"time"
)

type Blog struct {
	ID                int64     `yaml:"id"`
	Title             string    `yaml:"title"`
	Date              time.Time `yaml:"date"`
	Slug              string    `yaml:"slug"`
	Image             string    `yaml:"image,omitempty"`
	Excerpt           string    `yaml:"excerpt,omitempty"`
	Content           string    `yaml:"content"`
	Tags              []string  `yaml:"tags,omitempty"`
	Author            string    `yaml:"author,omitempty"`
	AuthorEmail       string    `yaml:"authorEmail,omitempty"`
	RelatedProducts   []int64   `yaml:"relatedProducts,omitempty"` // Assuming related products are referenced by IDs
	RelatedBundles    []int64   `yaml:"relatedBundles,omitempty"`  // Assuming related bundles are referenced by IDs
	IsVisible         bool      `yaml:"isVisible"`
	FeatureOnHomepage bool      `yaml:"featureOnHomepage,omitempty"`
	SEOTitle          string    `yaml:"SEOTitle,omitempty"`
	SEODescription    string    `yaml:"SEODescription,omitempty"`
	SEOKeywords       []string  `yaml:"SEOKeywords,omitempty"`
}
